test/functional/ex_cmds/mksession_spec.lua

```diff
@@ -46,4 +46,21 @@ describe(':mksession', function()
     command('tabnext 2')
     eq(cwd_dir .. get_pathsep() .. tab_dir, funcs.getcwd())
   end)
+
+	it('restores multiple windows with same terminal instances', function()
+		-- Create a view with two buffers referencing the same terminal instance
+		command('terminal')
+		command('split')
+		command('mksession ' .. session_file)
+
+		clear()
+
+		command('source ' .. session_file)
+		-- Getting the name of the buffer shown to compare with the other window
+		local bufname = funcs.bufname('%')
+
+		command('wincmd k')
+		eq(bufname, funcs.bufname('%'))
```

Comment:
Let's check `b:terminal_job_pid` instead, that will check the real PID instead of whatever happens to be the buffer name.

```lua
local expected_pid = helpers.eval('b:terminal_job_pid')
eq(1, funcs.winnr())
command('wincmd k')
eq(2, funcs.winnr())
eq(expected_pid, helpers.eval('b:terminal_job_pid'))
```

Reply:
I guess buffer id (`funcs.bufnr('%')`) would work as well. We just want to use something that wasn't possibly shuffled around by the session logic itself.

Reply:
Yeah it is better :-) I also used `exe nbr . "wincmd w"` to change windows to be sure to go to the right windows directly, hope it is good.